NullPointerException in explode to parquet

Using `591f7e6`, getting a NullPointer when trying to `explode` and then write to parquet.

```
Caused by: java.lang.NullPointerException
        at is.hail.keytable.KeyTable$$anonfun$59.apply(KeyTable.scala:536)
        at is.hail.keytable.KeyTable$$anonfun$59.apply(KeyTable.scala:534)
        at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
        at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
```

Example:
```
from hail import *
hc = HailContext()
output = 'gs://'  # set this to some output file
a_anns = [u'va.info.AC', u'va.info.AF', u'va.info.AS_FilterStatus']

def index_into_arrays(a_based_annotations, vep_root=None):
    annotations = []
    if a_based_annotations:
        for ann in a_based_annotations:
            annotations.append('%s = %s[va.aIndex - 1]' % (ann, ann))
    if vep_root:
        sub_fields = ['transcript_consequences', 'intergenic_consequences', 'motif_feature_consequences', 'regulatory_feature_consequences']
        annotations.extend(['%s.%s = %s.%s.filter(x => x.allele_num == va.aIndex)' % (vep_root, sub_field, vep_root, sub_field) for sub_field in sub_fields])
    return annotations

kt = (hc.read('gs://gnomad-public/release-170228/gnomad.exomes.r2.0.1.sites.autosomes.vds').split_multi()
 .annotate_variants_expr('va.info = select(va.info, AC, AF, AS_FilterStatus)')
 .annotate_variants_expr(index_into_arrays(a_anns, vep_root='va.vep'))
 .annotate_variants_expr(['va.filters = va.filters.mkString("|")',
                          'va.info.AS_FilterStatus = va.info.AS_FilterStatus.mkString("|")',
                          'va.vep = va.vep.transcript_consequences.map(x => drop(x, domains))'])
 .variants_keytable())

# kt.flatten().to_dataframe().write.parquet(output)  # works
kt.flatten().explode('va.vep').to_dataframe().write.parquet(output)  # fails
```

--- a/src/main/scala/is/hail/keytable/KeyTable.scala
+++ b/src/main/scala/is/hail/keytable/KeyTable.scala
@@ -534,7 +534,11 @@ case class KeyTable(hc: HailContext, rdd: RDD[Annotation],
 
     val explodedRDD = rdd.flatMap { a =>
       val row = a.asInstanceOf[Row].toSeq
-      for (element <- row(index).asInstanceOf[Iterable[_]]) yield row.updated(index, element)
+      val it = row(index)
+      if (it == null)
+        None
+      else
+        for (element <- row(index).asInstanceOf[Iterable[_]]) yield row.updated(index, element)
     }.map(Annotation.fromSeq)
 
     KeyTable(hc, explodedRDD, newSignature, keyNames)

--- a/src/main/scala/is/hail/keytable/KeyTable.scala
+++ b/src/main/scala/is/hail/keytable/KeyTable.scala
@@ -532,11 +532,12 @@ case class KeyTable(hc: HailContext, rdd: RDD[Annotation],
 
     val newSignature = signature.copy(fields = fields.updated(index, Field(columnName, explodeType, index)))
 
+    val empty = Iterable.empty[Seq[Any]]
     val explodedRDD = rdd.flatMap { a =>
       val row = a.asInstanceOf[Row].toSeq
       val it = row(index)
       if (it == null)
-        None
+        empty
       else
         for (element <- row(index).asInstanceOf[Iterable[_]]) yield row.updated(index, element)
     }.map(Annotation.fromSeq)

